### PR TITLE
Fix layer hygen unable to identify proper sources 

### PR DESCRIPTION
**Pull request by Clebal**
Error: "my source is specified as type: 'sql' but I get `The source has an unknown type error` when trying to create a layer from hygen

### CODE REVIEW

**Review by albertoarana — APPROVED** [review]

**Review by VictorVelarde — APPROVED** [review]
Please, add before a "Fix xxxxx" entry in the CHANGELOG, and then you're ready to go 🚀

**Comment by Clebal** [comment]
@VictorVelarde We're ready 🚀

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Not released
 - Add cypress e2e tests in CI [#194](https://github.com/CartoDB/carto-react-template/pull/194)
+- Remove case insensitive flag in source regex type check in layer generator [#198](https://github.com/CartoDB/carto-react-template/pull/198)
```
Comment (VictorVelarde) [inline CHANGELOG.md:5]: @Clebal That's the solution, but what error does it fix? can you rephrase it a bit?

Comment (Clebal) [reply CHANGELOG.md]: Me ha quedado un poco largo, pero bueno, se entiende @VictorVelarde.

File: hygen/_templates/layer/new/prompt.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ const prompt = async ({ prompter, args }) => {
 
   // Detect what kind of layer we need (CartoSQLLayer, CartoBQTilerLayer)
   const selectedSourceFileContent = readFile(`src/data/sources/${answers.source_file}.js`);
-  const res = /(?:type: ')(?<type>[\w]*)(?:')/gi.exec(selectedSourceFileContent);
+  const res = /(?:type: ')(?<type>sql|bq*)(?:')/g.exec(selectedSourceFileContent);
```
Comment (VictorVelarde) [inline hygen/_templates/layer/new/prompt.js:59]: @AdriSolid we will have to consider this if we make the 'bq' to 'bigquery' refactor later

Comment (Clebal) [reply hygen/_templates/layer/new/prompt.js:59]: Yep, that's right. Maybe it'd be nice in a near future to refactor hygen a bit to use the same constants to define source types, it's also in other generators.
